spring-cloud-generator/scripts/generate-showcase.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+# To VERIFY: ./scripts/generate-showcase.sh
+# To UPDATE: /scripts/generate-showcase.sh -u
+UPDATE=0
+while getopts u flag
+do
+    case "${flag}" in
+        u) UPDATE=1;;
+    esac
+done
+
+# If not set, assume working directory is spring-cloud-generator
+if [[ -z "$SPRING_GENERATOR_DIR" ]]; then
+  SPRING_GENERATOR_DIR=`pwd`
+fi
+SPRING_ROOT_DIR=${SPRING_GENERATOR_DIR}/..
+SHOWCASE_STARTER_OLD_DIR=${SPRING_GENERATOR_DIR}/showcase/showcase-spring-starter
+SHOWCASE_STARTER_NEW_DIR=${SPRING_GENERATOR_DIR}/showcase/showcase-spring-starter-generated
+
+# $1 - directory containing existing showcase-spring-starter (golden)
+# $2 - directory containing newly generated showcase-spring-starter
+function verify(){
+  OLD_DIR=$1
+  NEW_DIR=$2
+  SHOWCASE_STARTER_DIFF=$(diff -r ${NEW_DIR}/src/main ${OLD_DIR}/src/main)
+  SHOWCASE_STARTER_POM_DIFF=$(diff -r ${NEW_DIR}/pom.xml ${OLD_DIR}/pom.xml)
+  if [ "$SHOWCASE_STARTER_DIFF" != "" ] || [ "$SHOWCASE_STARTER_POM_DIFF" != "" ]
+  then
+      echo "Differences detected in generated showcase starter module: "
+      echo "Diff from src/main: "
+      echo $SHOWCASE_STARTER_DIFF
+      echo "Diff from pom.xml: "
+      echo $SHOWCASE_STARTER_POM_DIFF
+      exit 1;
+  else
+      echo "No differences found in showcase-spring-starter"
+      rm -r ${NEW_DIR}
+  fi
+}
+
+# $1 - target directory for generated starter
+function generate_showcase_spring_starter(){
+  # do something
+  SHOWCASE_STARTER_DIR=$1
+
+  # Compute the parent project version.
+  cd ${SPRING_ROOT_DIR}
+  PROJECT_VERSION=$(./mvnw help:evaluate -Dexpression=project.version -q -DforceStdout)
+  cd ${SPRING_GENERATOR_DIR}
+  GAPIC_GENERATOR_JAVA_VERSION=$(./../mvnw help:evaluate -Dexpression=gapic-generator-java-bom.version -q -DforceStdout)
+
+  # Install local snapshot jar for spring generator
+  cd ${SPRING_GENERATOR_DIR} && mvn install
+
+  # Clone sdk-platform-java (with showcase library)
+  git clone https://github.com/googleapis/sdk-platform-java.git
+  git checkout "v${GAPIC_GENERATOR_JAVA_VERSION}"
+
+  # Alternative considered: if showcase client library is available on Maven Central,
```

Comment:
> Is it because we have to rebuild Showcase from scratch or because we're not able to reuse the Spring CodeGen scripts?

@meltsufin it's because of the latter (+1 to @blakeli0's comment above). 

There are a number of nuances in bazel workspace modifications from the Spring Codegen scripts that rely on `googleapis` structure, and [lines 68-115](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1957/files#diff-8b45adb839b5d665b9765b501c4d48cfa965ad0612dcbd0c161ceebd5f38837dR68-R115) is a working attempt to achieve the sdk-platform-java/showcase equivalent for generating the spring starters). Now that #1956 is in, I'll take a look at refactoring this script on top (https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1957#discussion_r1232842089).

Reply:
> discussed this briefly in the past, both of us think it's a good idea.

It could be a good idea given the right use case. The previous use case was binary compatibility testing, but we've now implemented a local version of this check without needing a maven central artifact.

However, this use case would still require that we checkout the sdk-platform-java repo -- so I'm not sure publishing to maven central buys us anything significant.

Reply:
For a bit more context: currently the setup here checks out `sdk-platform-java` for two reasons:
1. To locally install the generated showcase client library (not available on Maven Central)
2. For leaning on bazel generation setup ([sdk-platform-java/WORKSPACE](https://github.com/googleapis/sdk-platform-java/blob/main/WORKSPACE), [sdk-platform-java/showcase/BUILD.bazel](https://github.com/googleapis/sdk-platform-java/blob/main/showcase/BUILD.bazel), and protos) instead of writing one from scratch

If having showcase available in Maven Central eliminates the step in 1, a feasible alternative to 2 (to avoid cloning sdk-platform-java) is to lean on googleapis’s bazel generation setup (which the Spring Codegen workflow for actual client libraries use) and follow something similar to [these steps](https://github.com/googleapis/sdk-platform-java/blob/autoconfig-gen-draft2/DEVELOPMENT.md#running-the-plugin) - copy protos from gapic-showcase, write a short BUILD.bazel, and make corresponding modifications to `googleapis/WORKSPACE`.

This is probably not worth the effort if the end goal is just to simplify this script, but could be more relevant if there is preference to avoid cloning sdk-platform-java altogether and reduce the number of repos that Spring Codegen locally depends on.